main.nf

```diff
@@ -463,6 +694,154 @@ if(!params.bed12){
+          publishDir path: { params.saveReference ? "${params.outdir}/reference_genome" : params.outdir },
+                     saveAs: { params.saveReference ? it : null }, mode: 'copy'
+
+          input:
+          file gtf from gtf_makeHisatSplicesites
+
+          output:
+          file "${gtf.baseName}.hisat2_splice_sites.txt" into indexing_splicesites, alignment_splicesites
+
+          script:
+          """
+          hisat2_extract_splice_sites.py $gtf > ${gtf.baseName}.hisat2_splice_sites.txt
+          """
+      }
+  }
+
+  /*
+   * PREPROCESSING - Build HISAT2 index
+   */
+  if(params.aligner == 'hisat2' && !params.hisat2_index && params.fasta){
+      process makeHISATindex {
+          tag "$fasta"
+          publishDir path: { params.saveReference ? "${params.outdir}/reference_genome" : params.outdir },
+                     saveAs: { params.saveReference ? it : null }, mode: 'copy'
+
+          input:
+          file fasta from ch_fasta_for_hisat_index
+          file indexing_splicesites from indexing_splicesites
+          file gtf from gtf_makeHISATindex
+
+          output:
+          file "${fasta.baseName}.*.ht2*" into hs2_indices
+
+          script:
+          if( !task.memory ){
+              log.info "[HISAT2 index build] Available memory not known - defaulting to 0. Specify process memory requirements to change this."
+              avail_mem = 0
+          } else {
+              log.info "[HISAT2 index build] Available memory: ${task.memory}"
+              avail_mem = task.memory.toGiga()
+          }
+          if( avail_mem > params.hisat_build_memory ){
+              log.info "[HISAT2 index build] Over ${params.hisat_build_memory} GB available, so using splice sites and exons in HISAT2 index"
+              extract_exons = "hisat2_extract_exons.py $gtf > ${gtf.baseName}.hisat2_exons.txt"
+              ss = "--ss $indexing_splicesites"
+              exon = "--exon ${gtf.baseName}.hisat2_exons.txt"
+          } else {
+              log.info "[HISAT2 index build] Less than ${params.hisat_build_memory} GB available, so NOT using splice sites and exons in HISAT2 index."
+              log.info "[HISAT2 index build] Use --hisat_build_memory [small number] to skip this check."
+              extract_exons = ''
+              ss = ''
+              exon = ''
+          }
+          """
+          $extract_exons
+          hisat2-build -p ${task.cpus} $ss $exon $fasta ${fasta.baseName}.hisat2_index
+          """
+      }
+  }
+}
+
+
+/*
+ * PREPROCESSING - Create Salmon transcriptome index
+ */
+if(params.pseudo_aligner == 'salmon' && !params.salmon_index){
+    if(!params.transcript_fasta) {
+        process transcriptsToFasta {
+            tag "$fasta"
+            publishDir path: { params.saveReference ? "${params.outdir}/reference_genome" : params.outdir },
+                               saveAs: { params.saveReference ? it : null }, mode: 'copy'
+
+            when:
+            !params.transcript_fasta
```

Comment:
why this statement, since we're checking that already with the if statement?

Reply:
Could be dropped - mind making a suggestion here too?